code/modules/mob/living/living_health_procs.dm

```diff
@@ -537,6 +537,10 @@
 		H.chestburst = 0
 		H.update_headshot_overlay() //They don't have their brains blown out anymore, if they did.
 
+	if(isxeno(src))
+		var/mob/living/carbon/xenomorph/xeno = src
+		xeno.clear_debuffs()
```

Comment:
Instead of this, have the status effect components register to the `COMSIG_LIVING_REJUVENATED` signal.

Reply:
For interference, you could update `/mob/living/carbon/xenomorph/rejuvenate` to handle it there